backend/vehicles/src/power-unit-types/power-unit-types.controller.ts

```diff
@@ -33,33 +35,61 @@ export class PowerUnitTypesController {
     isArray: true,
   })
   @Get()
-  findAll() {
-    return this.powerUnitTypesService.findAll();
+  async findAll():Promise<ReadPowerUnitTypeDto[]>  {
+    const powerUnitType = await this.powerUnitTypesService.findAll();
+    if (powerUnitType.length > 0)
+   {
+    return powerUnitType;
+   }else{
+    throw new CustomNotFoundException("Get all power unit failed.Power unit types do not exist in database yet",HttpStatus.NOT_FOUND)
+   }
   }
+  
 
   @ApiOkResponse({
     description: 'The Power Unit Type Resource',
     type: ReadPowerUnitTypeDto,
   })
   @Get(':typeCode')
-  findOne(@Param('typeCode') typeCode: string) {
-    return this.powerUnitTypesService.findOne(typeCode);
+  async findOne(@Param('typeCode') typeCode: string):Promise<ReadPowerUnitTypeDto> {
+    const powerUnitType = await this.powerUnitTypesService.findOne(typeCode);
+    if (powerUnitType)
+   {
+    return powerUnitType;
+   }else{
+    throw new CustomNotFoundException("Get power unit failed. Power unit with typeCode "+typeCode+" does not exist in database",HttpStatus.NOT_FOUND)
```

Comment:
I personally dislike this error message. I think it reveals too much. A generic data not found would suffice.